### PR TITLE
correctly handle nccl for nvidia_gpu_device

### PR DESCRIPTION
**Pull request by cloudhan**
This is an ongoing effort of https://github.com/google/jax/pull/4843

nccl is not available on windows, `if_cuda` is not reliable.

### CODE REVIEW

**Comment by cloudhan** [comment]
@hawkinsp I think I should be mentioned here.

**Review by hawkinsp — APPROVED** [review]

**Comment by rthadur** [comment]
@cloudhan can you please check sanity build failures ?

**Comment by cloudhan** [comment]
@rthadur done

**Review by hawkinsp — APPROVED** [review]

**Comment by cloudhan** [comment]
Why it is blocked?

File: tensorflow/compiler/xla/pjrt/BUILD [hunk 1]
```diff
@@ -1,6 +1,5 @@
 load("//tensorflow/core/platform:rules_cc.bzl", "cc_library")
-load("//tensorflow:tensorflow.bzl", "tf_cc_test")
-load("@local_config_cuda//cuda:build_defs.bzl", "if_cuda")
+load("//tensorflow:tensorflow.bzl", "if_nccl", "tf_cc_test")
```
Comment (hawkinsp) [inline tensorflow/compiler/xla/pjrt/BUILD:2]: I'm not sure (I can't find the error from copybara), but I'm guessing this line is the problem. As part of our internal CI these imports get rewritten, and I think the rewrite expects only one thing is loaded per load line. Can you try replacing this with:
```
load("//tensorflow:tensorflow.bzl", "if_nccl")
load("//tensorflow:tensorflow.bzl", "tf_cc_test")
```

?

If that doesn't work, I'll make the same change myself on your behalf to get you unblocked.

Comment (cloudhan) [reply tensorflow/compiler/xla/pjrt/BUILD:2]: Hmm, without changing anything, copybara passed. So should I edit as suggested or just wait?

Comment (hawkinsp) [reply tensorflow/compiler/xla/pjrt/BUILD:2]: I think that change is still needed; it won't hurt either way, so I suggest making it and then hopefully the merge will go through.

**Review by hawkinsp — APPROVED** [review]

**Comment by hawkinsp** [comment]
@cloudhan Looks like that did the trick!

**Pull request merged**